tests/unit/clients/requests_tests.py

```diff
@@ -0,0 +1,125 @@
+from contextlib import nullcontext as does_not_raise
+from unittest import mock
+
+import pytest
+
+from prometheus_client import REGISTRY
+from requests import Request
+from requests import Response
+from requests import Session
+
+from baseplate.clients.requests import ACTIVE_REQUESTS
+from baseplate.clients.requests import BaseplateSession
+from baseplate.clients.requests import LATENCY_SECONDS
+from baseplate.clients.requests import REQUESTS_TOTAL
+from baseplate.lib.prometheus_metrics import getHTTPSuccessLabel
+
+
+@pytest.fixture
+def baseplate_session(request):
+    yield BaseplateSession(
+        adapter=mock.MagicMock(),
+        name="session_name",
+        span=mock.MagicMock(),
+        client_name=request.param,
+    )
+
+
+class TestBaseplateSessionProm:
+    def setup(self):
+        ACTIVE_REQUESTS.clear()
+        LATENCY_SECONDS.clear()
+        REQUESTS_TOTAL.clear()
+
+    # need to patch Session() somehow...
```

Comment:
It seems like these docs are maybe out of date? It seems like you figured out how to patch the session.

Reply:
yup leaving comments to myself 😅